packages/core/src/core.ts

```diff
@@ -199,17 +231,28 @@ export function debug(message: string): void {
 /**
  * Adds an error issue
  * @param message error issue message. Errors will be converted to string via toString()
+ * @param properties optional properties to add to the annotation. 
  */
-export function error(message: string | Error): void {
-  issue('error', message instanceof Error ? message.toString() : message)
+export function error(message: string | Error, properties: AnnotationProperties = {}): void {
+  issueCommand('error', toCommandProperties(properties), message instanceof Error ? message.toString() : message)
 }
 
 /**
- * Adds an warning issue
+ * Adds a warning issue
  * @param message warning issue message. Errors will be converted to string via toString()
+ * @param properties optional properties to add to the annotation. 
+ */
+export function warning(message: string | Error, properties: AnnotationProperties = {}): void {
+  issueCommand('warning', toCommandProperties(properties), message instanceof Error ? message.toString() : message)
+}
+
+/**
+ * Adds a notice issue
+ * @param message notice issue message. Errors will be converted to string via toString()
+ * @param properties optional properties to add to the annotation. 
  */
-export function warning(message: string | Error): void {
-  issue('warning', message instanceof Error ? message.toString() : message)
+export function notice(message: string | Error, properties: AnnotationProperties = {}): void {
```

Comment:
I'm trying to decide what to do with the `info` message below. Should `info` proxy to `notice`? I'm thinking maybe not because we wouldn't necessarily want existing actions to submit large numbers of annotations if they make heavy use of `info`